src/AGTHARN/FlyPE/Main.php

```diff
@@ -27,319 +27,177 @@
-				$event->setCancelled();
-			}
+		});
+		
+		/** @phpstan-ignore-next-line */
+		if ($this->getConfig()->get("enable-fly-ui") === true && $this->getConfig()->get("pay-for-fly") === true && $this->getConfig()->get("custom-ui-texts") === false) {
+			$cost = $this->getConfig()->get("buy-fly-cost");
+					
+			$form->setTitle("§l§7< §2FlyUI §7>");
+			$form->addButton("§aToggle Fly §e(Costs $ {$cost})");
+			$form->addButton("§cExit");
+			$form->sendToPlayer($player);
+			return $form;
+		} elseif ($this->getConfig()->get("enable-fly-ui") === true && $this->getConfig()->get("pay-for-fly") === false && $this->getConfig()->get("custom-ui-texts") === false) {
+			$form->setTitle("§l§7< §6FlyUI §7>");
+			$form->addButton("§aToggle Fly");
+			$form->addButton("§cExit");
+			$form->sendToPlayer($player);
+			return $form;
+		} elseif ($this->getConfig()->get("custom-ui-texts") === true) {
+			$cost = $this->getConfig()->get("buy-fly-cost");
+
+			$form->setTitle($this->getConfig()->get("fly-ui-title"));
+			$form->addButton(str_replace("{cost}", $cost, $this->getConfig()->get("fly-ui-toggle")));
+			$form->addButton($this->getConfig()->get("fly-ui-exit"));
+			$form->sendToPlayer($player);
+			return $form;
 		}
 	}
 	
-	public function onBlockPlace(BlockPlaceEvent $event){
-		$player = $event->getPlayer();
-		if(!$player instanceof Player) return;
-		if($player->getGamemode() === Player::CREATIVE) return;
-		if($this->getConfig()->get("block-placing") === false){
-			if($player->getAllowFlight() === true){
-				$event->setCancelled();
-			}
+	/**
+	 * doLevelChecks
+	 *
+	 * @param  Player $player
+	 * @return bool
+	 */
+	public function doLevelChecks(Player $player): bool {
+		$levelName = $player->getLevel()->getName();
+		$name = $player->getName();
+
+		if ($player->getGamemode() === Player::CREATIVE && $player->getAllowFlight() === true) {
+			$player->sendMessage(C::RED . str_replace("{name}", $name, $this->getConfig()->get("disable-fly-creative")));
+			return false;
 		}
+
+		if ($this->getConfig()->get("mode") === "blacklist" && !in_array($player->getLevel()->getName(), $this->getConfig()->get("blacklisted-worlds"))) {
+			return true;
+		}
+		if ($this->getConfig()->get("mode") === "whitelist" && in_array($player->getLevel()->getName(), $this->getConfig()->get("whitelisted-worlds"))) {
+			return true;
+		}
+		$player->sendMessage(C::RED . str_replace("{world}", $levelName, $this->getConfig()->get("flight-not-allowed")));
+		return false;
 	}
 	
-	public function onPlayerItemConsume(PlayerItemConsumeEvent $event){
-		$player = $event->getPlayer();
-		if(!$player instanceof Player) return;
-		if($player->getGamemode() === Player::CREATIVE) return;
-		if($this->getConfig()->get("player-eating") === false){
-			if($player->getAllowFlight() === true){
-				$event->setCancelled();
-			}
+	/**
+	 * toggleFlight
+	 *
+	 * @param  Player $player
+	 * @return void
+	 */
+	public function toggleFlight(Player $player): void {
```

Comment:
Perhaps you could add another parameter here that takes a string with either `enable` or `disable`, so you could be in more control, and not relying on if they are already flying or not. Just make sure you change the code in this method a little bit to match it.

Reply:
I feel that keeping it is better. I would have to add even more checks if I use `enable` or `disable` instead. However, I may consider doing that if I am adding an on/off button to the FlyUI instead of a toggle button.